### PR TITLE
Move Custom Resources back to istio-system

### PR DESCRIPTION
**Pull request by andraxylia**
Move CRs for global configs back to istio-system and discard istio-config-default namespace.

Start mixer with --configDefaultNamespace=istio-system.

### CODE REVIEW

**Comment by kyessenov** [comment]
cc'ed release tzars.

File: install/kubernetes/istio-auth.yaml [hunk 1]
```diff
@@ -273,6 +273,7 @@ spec:
         args:
           - --configStoreURL=fs:///etc/opt/mixer/configroot
           - --configStore2URL=k8s://
+          - --configDefaultNamespace=istio-system
           - --logtostderr
           - -v
           - "4"
```
Comment (kyessenov) [inline install/kubernetes/istio-auth.yaml]: is it worth turning down logging? doesn't v4 cause unnecessary performance burden?

File: install/kubernetes/istio-auth.yaml [hunk 2]
```diff
@@ -817,11 +813,11 @@ apiVersion: "config.istio.io/v1alpha2"
 kind: prometheus
 metadata:
   name: handler
-  namespace: istio-config-default
+  namespace: istio-system
 spec:
   metrics:
   - name: request_count
-    instance_name: requestcount.metric.istio-config-default
+    instance_name: requestcount.metric.istio-system
```
Comment (kyessenov) [inline install/kubernetes/istio-auth.yaml]: @mandarjog can the namespace domain be inferred here?

File: tests/e2e/framework/kubernetes.go [hunk 3]
```diff
@@ -36,7 +36,6 @@ const (
 	nonAuthInstallFile = "istio-one-namespace.yaml"
 	authInstallFile    = "istio-one-namespace-auth.yaml"
 	istioSystem        = "istio-system"
-	mixerConfigDefault = "istio-config-default"
```
Comment (kyessenov) [inline tests/e2e/framework/kubernetes.go]: can we keep this around and set same value as istio-system? seems like more future-proof?

File: install/kubernetes/istio-auth.yaml (hunk 1 above)
Comment (douglas-reid) [reply install/kubernetes/istio-auth.yaml]: probably. but let's fix that in a different PR.

File: install/kubernetes/istio-auth.yaml (hunk 2 above)
Comment (douglas-reid) [reply install/kubernetes/istio-auth.yaml]: it cannot at this point. this is a code change to the prometheus adapter (or elsewhere in Mixer).

File: install/kubernetes/istio-auth.yaml (hunk 1 above)
Comment (rshriram) [reply install/kubernetes/istio-auth.yaml]: You sure? One of our benchmarks showed a significant slowdown in perf when we turned on logging. But that was for pilot. I am just extrapolating.

Comment (ZackButcher) [reply install/kubernetes/istio-auth.yaml]: Mixer logs a lot.

Comment (douglas-reid) [reply install/kubernetes/istio-auth.yaml]: My concern is just that we might want to wait to turn down logging once things have settled a bit more and we are closer to release.  Prod deployments of Mixer should probably set v=0.

Comment (ZackButcher) [reply install/kubernetes/istio-auth.yaml]: True - I guess lets keep it verbose till we get closer to the final 0.2 cut.

File: install/kubernetes/istio-auth.yaml (hunk 2 above)
Comment (douglas-reid) [reply install/kubernetes/istio-auth.yaml]: https://github.com/istio/mixer/issues/1323

**Comment by rshriram** [comment]
LGTM sans the istio-config-default

File: install/kubernetes/istio-auth.yaml (hunk 1 above)
Comment (ldemailly) [reply install/kubernetes/istio-auth.yaml]: can we change the flag default value too to be `istio-system`

File: tests/e2e/framework/kubernetes.go (hunk 3 above)
Comment (andraxylia) [reply tests/e2e/framework/kubernetes.go]: It's not hard to put back, if needed. Let's try to keep the code clean, otherwise we might not get back to it.

**Comment by ldemailly** [comment]
need to make sure the tests pass

also like the other pr make sure
```
git grep istio-config-default
```
is empty

**Comment by mandarjog** [comment]
@andraxylia looks like this is passing tests now with the Mixer fix.

/lgtm

**Pull request merged**